doc/developer/platform/formalism.md

```diff
@@ -0,0 +1,176 @@
+
+*   `Create(id, description) -> (ReadCapability, WriteCapability)`: binds to `id` the TVC described by `description`.
+
+    The command returns capabilities naming `id`, with frontiers set to initial values chosen by the Storage layer.
+
+    The collection associated with `id` is based on `description`, but it is up to the Storage layer to determine what the source's contents will be.
+    A standarad example is a Kafka topic, whose contents are "added" to the underlying collection as they are observed.
+    The Storage layer only needs to ensure that it can reliably produce the updates between `since` and `upper`, and otherwise the nature of the content is up for discussion.
+    It is an error to re-use a previously used `id`.
+
+*   `SubscribeAt(ReadCapability(id, frontier))`: returns a snapshot of `id` at `frontier`, followed by an ongoing stream of subsequent updates.
+
+    This command returns the contents of `id` as of `frontier` once they are known, and updates thereafter once they are known.
+    The snapshot and stream contain in-line statements of the subscription's `upper` frontier: those times for which updates may still arrive.
+    This call may block, or not be promptly responded to, if `frontier` is greater or equal to the current `upper` of `id`.
+    The subscription can be canceled by either endpoint, and the recipient should only downgrade their read capability when they are certain they have all data for an interval.
+
+    A subscription can be constructed with additional arguments that change how the data is returned to the user.
+    For example, the user may ask to not receive the initial snapshot, rather that receive and then discard it.
+    For example, the user may provide filtering and projection that can be applied before the data are transmitted.
+
+*   `UpdateAndDowngrade(WriteCapability(id, frontier), updates, new_frontier)`: applies `updates` to `id` and downgrades `frontier` to `new_frontier`.
+
+    All times in `updates` must be greater or equal to `frontier` and not greater or equal to `new_frontier`.
+
+    This is the analog of the `INSERT` statement, though `updates` are signed and general enough to support `DELETE` and `UPDATE` at the same time.
+
+*   `AcquireCapabilities(id) -> (ReadCapability, WriteCapability)`: provides capabilities for `id` at its current `since` and `upper` frontiers.
+
+    This method is a best-effort attempt to regain control of the frontiers of a collection.
+    Its most common uses are to recover capabilities that have expired (leases) or to attempt to read a TVC that one did not create (or otherwise receive capabilities for).
+    If the frontiers have been fully released by all other parties, this call may result in empty frontiers (useless).
+
+The Storage layer provides its output through `SubscribeAt` and `AcquireCapabilities`, which reveal the contents and frontiers of the TVC.
```

Comment:
nit: this might be better placed at the beginning of this section, right after

> The Storage layer presents a narrow API about which it makes guarantees.

as it provides high-level information that is useful when reading through the API details.